### PR TITLE
CNV-39781: lldp vlan and system name filters

### PR DESCRIPTION
**Pull request by upalatucci**
**All filters**

Screenshot from 2024-03-25 15-36-59

**VLAN filter**
Screenshot from 2024-03-25 15-37-46

**System name filter**
Screenshot from 2024-03-25 15-37-08

### CODE REVIEW

File: src/views/states/list/hooks/useStateFilters.ts [hunk 1]
```diff
@@ -38,6 +43,36 @@ export const useStateSearchFilters = (): RowSearchFilter<V1beta1NodeNetworkState
...
+      placeholder: t('Search by VLAN name...'),
+    },
+    {
+      type: FILTER_TYPES.LLDP_SYSTEM_NAME,
+      filter: (searchText, obj) => {
+        const searchSystemName = searchText?.selected?.[0];
+        if (isEmpty(searchSystemName)) return true;
+
+        const interfaces = obj?.status?.currentState
+          ?.interfaces as NodeNetworkConfigurationInterface[];
```
Comment (hstastna) [inline src/views/states/list/hooks/useStateFilters.ts]: Here I can see some repeating code. Maybe we could create some getter or some helper function to simply get what we want from the object? `getInterfaces(obj)` maybe? Then you could use it where needed and the code could look nicer.

Comment (upalatucci) [reply src/views/states/list/hooks/useStateFilters.ts]: good point

File: src/views/states/list/hooks/useStateFilters.ts [hunk 2]
```diff
@@ -30,14 +35,41 @@ export const useStateSearchFilters = (): RowSearchFilter<V1beta1NodeNetworkState
...
       filterGroupName: t('MAC address'),
       placeholder: t('Search by MAC address...'),
     },
+    {
+      type: FILTER_TYPES.LLDP_NAME,
+      filter: (searchText, obj) => {
+        const searchLLDPName = searchText?.selected?.[0];
+        if (isEmpty(searchLLDPName)) return true;
+
+        const interfaces = getInterfaces(obj);
```
Comment (hstastna) [inline src/views/states/list/hooks/useStateFilters.ts:51]: Nice to define and use `getInterfaces` here and in other places to simplify the code a bit :) 👍🏽

**Review by hstastna — APPROVED** [review]
Nice work, Ugo! 🎇

**Pull request merged**